scripts/calculateKpi/protocols/beefy/index.ts

```diff
@@ -126,6 +164,53 @@ export async function calculateVaultRevenue(
   return totalUsdContribution
 }
 
+/**
+ * Calculates vault management fee revenue generated for Beefy Finance protocol.
+ *
+ * **KPI Unit**: USD (United States Dollars)
+ *
+ * **Business Purpose**: Measures the vault management fee revenue attributable to a specific user's
+ * deposit activity in Beefy vaults. This metric quantifies the economic value generated by user
+ * participation in Beefy's yield optimization strategies and supports fee distribution calculations.
+ *
+ * **Protocol Context**: Beefy Finance is a decentralized yield optimization platform that automatically
+ * compounds yields for users across multiple DeFi protocols. It charges a performance fee for managing
+ * user deposits and generating optimized returns.
+ *
+ * **Supported Networks**: Multiple EVM-compatible networks where Beefy has deployed vault strategies
+ *
+ * **Data Sources**:
+ * - **Beefy API**: Vault configurations, TVL data, and strategy metadata from `api.beefy.finance`
+ * - **RPC Queries**: User deposit/withdrawal events via Viem public client calls to vault contracts
+ * - **Token Price API**: Historical token prices via `fetchTokenPrices` utility for USD conversion
+ * - **Block Data**: Timestamps via `getBlockRange` utility for temporal filtering
+ * - **Network RPCs**: Multi-network vault contract interactions across Arbitrum, Polygon, Base, Optimism
+ *
+ * **Business Assumptions**:
+ * - Standard management fee rate is 4.5% of total vault yield
+ * - Revenue attribution is proportional to user's share of vault deposits within time window
+ * - USD conversion uses token prices at time of each transaction
+ * - Only active vault strategies with positive TVL are included
+ * - Fee distribution is immediate and based on real-time vault performance
+ *
+ * **Fee Structure**: 4.5% performance fee on all vault yields (industry standard for yield aggregators)
+ *
+ * **Calculation Method**:
+ * 1. Retrieves active vault configurations from Beefy API
+ * 2. Filters vaults by supported networks and active status
+ * 3. For each vault, queries user's deposit/withdrawal events within time window
```

Comment:
```suggestion
 * 3. For each vault, queries user's deposit/withdrawal events within time window
 * 4. For each vault, queries historical Fee events within time window
 * 5. Calculates user's proportional share of vault deposits at the time of each Fee event
 * 6. Determines user contribution to management fee based off their proportional share of vault TVL
 * 7. Converts to USD using historical token prices at transaction timestamps
 * 8. Aggregates user fee contribution across all vaults and networks to get total user fee contribution
 ```

This is from memory, but I believe this is more correct.